De-globify ocean analysis yamls

AndrewEichmann-NOAA opened the pull request:
# Description

Removes the *yaml glob used in archiving of ocean analysis files and replaces them with explicit filenames, to avoid sweeping in unrelated yaml files.

Resolves https://github.com/NOAA-EMC/global-workflow/issues/4158


# Type of change
- [x] Bug fix (fixes something broken)
- [ ] New feature (adds functionality)
- [ ] Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO
- Does this change require an update to any of the following submodules? NO (If YES, please add a link to any PRs that are pending.)
  - [ ] EMC verif-global <!-- NOAA-EMC/EMC_verif-global#1234 -->
  - [ ] GDAS <!-- NOAA-EMC/GDASApp#1234 -->
  - [ ] GFS-utils <!-- NOAA-EMC/gfs-utils#1234 -->
  - [ ] GSI <!-- NOAA-EMC/GSI#1234 -->
  - [ ] GSI-monitor <!-- NOAA-EMC/GSI-Monitor#1234 -->
  - [ ] GSI-utils <!-- NOAA-EMC/GSI-Utils#1234 -->
  - [ ] UFS-utils <!-- ufs-community/UFS_UTILS#1234 -->
  - [ ] UFS-weather-model <!-- ufs-community/ufs-weather-model#1234 -->
  - [ ] wxflow <!-- NOAA-EMC/wxflow#1234 -->

# How has this been tested?

~~Not yet tested~~
Tested with `C96C48mx500_S2SW_cyc_gfs` and `C48mx500_3DVarAOWCDA` on Ursa, with archived tarballs as follows:

```
[Andrew.Eichmann@ufe03 hpss-check]$ htar -t -f /NCEPDEV/emc-global/1year/Andrew.Eichmann/URSA/scratch/noglobvar/2021032500//gfsocean_analysis.tar  | grep yaml
[connecting to hpsscore1.fairmont.rdhpcs.noaa.gov/1217]
HTAR: -rw-r--r--  Andrew.Eichmann/da       8834 2025-10-23 15:07  gfs.20210325/00/conf/gfs.t00z.bmat_fields_metadata.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da        143 2025-10-29 14:30  gfs.20210325/00/conf/gfs.t00z.gridgen.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da        773 2025-10-29 14:30  gfs.20210325/00/conf/gfs.t00z.soca_chgres.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da       1026 2025-10-29 14:30  gfs.20210325/00/conf/gfs.t00z.soca_diagb.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da        573 2025-10-29 14:30  gfs.20210325/00/conf/gfs.t00z.soca_setcorscales.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da        363 2025-10-29 14:30  gfs.20210325/00/conf/gfs.t00z.soca_vtscales.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da      16813 2025-10-29 14:40  gfs.20210325/00/conf/var.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da       1020 2025-10-29 14:30  gfs.20210325/00/conf/gfs.t00z.soca_parameters_diffusion_vt.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da       1062 2025-10-29 14:30  gfs.20210325/00/conf/gfs.t00z.soca_parameters_diffusion_hz.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da       2626 2025-10-29 14:40  gfs.20210325/00/conf/soca_incpostproc.yaml
[Andrew.Eichmann@ufe03 hpss-check]$ htar -t -f /NCEPDEV/emc-global/1year/Andrew.Eichmann/URSA/scratch/noglobvar/2021032500//gdasocean_analysis.tar  | grep yaml
[connecting to hpsscore1.fairmont.rdhpcs.noaa.gov/1217]
HTAR: -rw-rw-r--  Andrew.Eichmann/da        573 2025-10-29 14:30  gdas.20210325/00/conf/gdas.t00z.soca_setcorscales.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da        773 2025-10-29 14:30  gdas.20210325/00/conf/gdas.t00z.soca_chgres.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da       1026 2025-10-29 14:30  gdas.20210325/00/conf/gdas.t00z.soca_diagb.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da        143 2025-10-29 14:30  gdas.20210325/00/conf/gdas.t00z.gridgen.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da        363 2025-10-29 14:30  gdas.20210325/00/conf/gdas.t00z.soca_vtscales.yaml
HTAR: -rw-r--r--  Andrew.Eichmann/da       8834 2025-10-23 15:07  gdas.20210325/00/conf/gdas.t00z.bmat_fields_metadata.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da      16834 2025-10-29 14:40  gdas.20210325/00/conf/var.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da       1062 2025-10-29 14:30  gdas.20210325/00/conf/gdas.t00z.soca_parameters_diffusion_hz.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da       1020 2025-10-29 14:30  gdas.20210325/00/conf/gdas.t00z.soca_parameters_diffusion_vt.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da       2626 2025-10-29 14:40  gdas.20210325/00/conf/soca_incpostproc.yaml
[Andrew.Eichmann@ufe03 hpss-check]$ htar -t -f /NCEPDEV/emc-global/1year/Andrew.Eichmann/URSA/scratch/noglob/2021122018/gdasocean_analysis.tar | grep yaml
[connecting to hpsscore1.fairmont.rdhpcs.noaa.gov/1217]
HTAR: -rw-rw-r--  Andrew.Eichmann/da       2628 2025-10-29 17:15  gdas.20211220/18/conf/gdas.t18z.soca_ensb.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da        785 2025-10-29 17:15  gdas.20211220/18/conf/gdas.t18z.soca_ensweights.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da        573 2025-10-29 17:15  gdas.20211220/18/conf/gdas.t18z.soca_setcorscales.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da        773 2025-10-29 17:15  gdas.20211220/18/conf/gdas.t18z.soca_chgres.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da       1026 2025-10-29 17:15  gdas.20211220/18/conf/gdas.t18z.soca_diagb.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da        143 2025-10-29 17:15  gdas.20211220/18/conf/gdas.t18z.gridgen.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da        363 2025-10-29 17:15  gdas.20211220/18/conf/gdas.t18z.soca_vtscales.yaml
HTAR: -rw-r--r--  Andrew.Eichmann/da       8834 2025-10-23 15:07  gdas.20211220/18/conf/gdas.t18z.bmat_fields_metadata.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da      12819 2025-10-29 17:25  gdas.20211220/18/conf/var.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da       1062 2025-10-29 17:15  gdas.20211220/18/conf/gdas.t18z.soca_parameters_diffusion_hz.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da       1020 2025-10-29 17:15  gdas.20211220/18/conf/gdas.t18z.soca_parameters_diffusion_vt.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da       2626 2025-10-29 17:25  gdas.20211220/18/conf/soca_incpostproc.yaml
[Andrew.Eichmann@ufe03 hpss-check]$ htar -t -f /NCEPDEV/emc-global/1year/Andrew.Eichmann/URSA/scratch/noglob/2021122018/gfsocean_analysis.tar | grep yaml
[connecting to hpsscore1.fairmont.rdhpcs.noaa.gov/1217]
HTAR: -rw-r--r--  Andrew.Eichmann/da       8834 2025-10-23 15:07  gfs.20211220/18/conf/gfs.t18z.bmat_fields_metadata.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da        143 2025-10-29 17:15  gfs.20211220/18/conf/gfs.t18z.gridgen.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da        773 2025-10-29 17:15  gfs.20211220/18/conf/gfs.t18z.soca_chgres.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da       1026 2025-10-29 17:15  gfs.20211220/18/conf/gfs.t18z.soca_diagb.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da        573 2025-10-29 17:15  gfs.20211220/18/conf/gfs.t18z.soca_setcorscales.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da        363 2025-10-29 17:15  gfs.20211220/18/conf/gfs.t18z.soca_vtscales.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da      12804 2025-10-29 17:25  gfs.20211220/18/conf/var.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da       1062 2025-10-29 17:15  gfs.20211220/18/conf/gfs.t18z.soca_parameters_diffusion_hz.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da       1020 2025-10-29 17:15  gfs.20211220/18/conf/gfs.t18z.soca_parameters_diffusion_vt.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da       2626 2025-10-29 17:25  gfs.20211220/18/conf/soca_incpostproc.yaml
[Andrew.Eichmann@ufe03 hpss-check]$ htar -t -f /NCEPDEV/emc-global/1year/Andrew.Eichmann/URSA/scratch/noglob/2021122100/gdasocean_analysis.tar | grep yaml
[connecting to hpsscore1.fairmont.rdhpcs.noaa.gov/1217]
HTAR: -rw-rw-r--  Andrew.Eichmann/da       2628 2025-10-29 18:15  gdas.20211221/00/conf/gdas.t00z.soca_ensb.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da        785 2025-10-29 18:15  gdas.20211221/00/conf/gdas.t00z.soca_ensweights.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da        573 2025-10-29 18:15  gdas.20211221/00/conf/gdas.t00z.soca_setcorscales.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da        773 2025-10-29 18:15  gdas.20211221/00/conf/gdas.t00z.soca_chgres.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da       1026 2025-10-29 18:15  gdas.20211221/00/conf/gdas.t00z.soca_diagb.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da        143 2025-10-29 18:15  gdas.20211221/00/conf/gdas.t00z.gridgen.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da        363 2025-10-29 18:15  gdas.20211221/00/conf/gdas.t00z.soca_vtscales.yaml
HTAR: -rw-r--r--  Andrew.Eichmann/da       8834 2025-10-23 15:07  gdas.20211221/00/conf/gdas.t00z.bmat_fields_metadata.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da      15001 2025-10-29 18:25  gdas.20211221/00/conf/var.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da       1062 2025-10-29 18:15  gdas.20211221/00/conf/gdas.t00z.soca_parameters_diffusion_hz.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da       1020 2025-10-29 18:15  gdas.20211221/00/conf/gdas.t00z.soca_parameters_diffusion_vt.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da       2626 2025-10-29 18:25  gdas.20211221/00/conf/soca_incpostproc.yaml
[Andrew.Eichmann@ufe03 hpss-check]$ htar -t -f /NCEPDEV/emc-global/1year/Andrew.Eichmann/URSA/scratch/noglob/2021122100/gfsocean_analysis.tar | grep yaml
[connecting to hpsscore1.fairmont.rdhpcs.noaa.gov/1217]
HTAR: -rw-r--r--  Andrew.Eichmann/da       8834 2025-10-23 15:07  gfs.20211221/00/conf/gfs.t00z.bmat_fields_metadata.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da        143 2025-10-29 18:11  gfs.20211221/00/conf/gfs.t00z.gridgen.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da        773 2025-10-29 18:11  gfs.20211221/00/conf/gfs.t00z.soca_chgres.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da       1026 2025-10-29 18:11  gfs.20211221/00/conf/gfs.t00z.soca_diagb.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da        573 2025-10-29 18:11  gfs.20211221/00/conf/gfs.t00z.soca_setcorscales.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da        363 2025-10-29 18:11  gfs.20211221/00/conf/gfs.t00z.soca_vtscales.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da      14983 2025-10-29 18:20  gfs.20211221/00/conf/var.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da       1062 2025-10-29 18:11  gfs.20211221/00/conf/gfs.t00z.soca_parameters_diffusion_hz.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da       1020 2025-10-29 18:11  gfs.20211221/00/conf/gfs.t00z.soca_parameters_diffusion_vt.yaml
HTAR: -rw-rw-r--  Andrew.Eichmann/da       2626 2025-10-29 18:20  gfs.20211221/00/conf/soca_incpostproc.yaml
[Andrew.Eichmann@ufe03 hpss-check]$
```

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have documented my code, including function, input, and output descriptions
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] This change is covered by an existing CI test or a new one has been added
- [x] Any new scripts have been added to the .github/CODEOWNERS file with owners
- [x] I have made corresponding changes to the system documentation if necessary

AndrewEichmann-NOAA committed:
De-globify ocean analysis yamls

removes the *yaml glob used in archiving and replace with explicit filesnames

JessicaMeixner-NOAA commented:
Thanks for making this PR @AndrewEichmann-NOAA !

AndrewEichmann-NOAA committed:
Merge branch 'develop' into bugfix/superfluous-yamls

JessicaMeixner-NOAA commented on parm/archive/gdasocean_analysis.yaml.j2:
```diff
@@ -37,7 +37,18 @@ gdasocean_analysis:
+        - '{{ COMIN_CONF | relpath(ROTDIR) }}/{{ head }}soca_diagb.yaml'
+        - '{{ COMIN_CONF | relpath(ROTDIR) }}/{{ head }}gridgen.yaml'
+        - '{{ COMIN_CONF | relpath(ROTDIR) }}/{{ head }}soca_vtscales.yaml'
+        - '{{ COMIN_CONF | relpath(ROTDIR) }}/{{ head }}soca_ensweights.yaml'
```
Does this config only exist for certain options? 

See: vi /6/ira-sti/world-shared/Jessica.Meixner/togwdevrt17/hpss01/COMROOT/C48mx500_3DVarAOWCDA_hpss01/logs/2021032500/gdas_arch_tar_gdasocean_analysis.log.0 where we get an error that this file does not exist. 

In the conf file I have: 
```
backup_tarballs.yaml		     gdas.t00z.soca_chgres.yaml			  gdas.t00z.soca_vtscales.yaml	ufs.ice_in	     ufs.ww3_shel.nml
fields_metadata.yaml		     gdas.t00z.soca_diagb.yaml			  MOM_parameter_doc.all		ufs.input.nml	     var.yaml
gdas.t00z.bmat_fields_metadata.yaml  gdas.t00z.soca_parameters_diffusion_hz.yaml  obsop_name_map.yaml		ufs.model_configure
gdas.t00z.gridgen.yaml		     gdas.t00z.soca_parameters_diffusion_vt.yaml  soca_incpostproc.yaml		ufs.MOM_input
gdas.t00z.obsop_name_map.yaml	     gdas.t00z.soca_setcorscales.yaml		  ufs.diag_table		ufs.ufs.configure
```

AndrewEichmann-NOAA replied:
@JessicaMeixner-NOAA Which file doesn't exist?

JessicaMeixner-NOAA replied:
soca_ensweights.yam

AndrewEichmann-NOAA replied:
That does sound hybrid-specifc. @guillaumevernieres ?

AndrewEichmann-NOAA replied:
Looks like it - that and `soca_ensb` - this is what gets saved to com:

https://github.com/NOAA-EMC/GDASApp/blob/37522a915df82b439eb22458f6403dfa6021c29a/parm/marine/marine_bmat_save.yaml.j2#L36-39

guillaumevernieres replied:
yes, `soca_ensweights.yaml` is only for the hybrid.

AndrewEichmann-NOAA committed:
allowances for hyb/3var

AndrewEichmann-NOAA commented:
Testing help up by /scratch4 quota

AndrewEichmann-NOAA committed:
Merge branch 'develop' into bugfix/superfluous-yamls

AndrewEichmann-NOAA committed:
added gfs archiving

AndrewEichmann-NOAA committed:
Merge branch 'develop' into bugfix/superfluous-yamls

aerorahul commented on parm/archive/gdasocean_analysis.yaml.j2:
```diff
@@ -33,11 +33,22 @@ gdasocean_analysis:
+        - '{{ COMIN_CONF | relpath(ROTDIR) }}/{{ head }}soca_vtscales.yaml'
+        - '{{ COMIN_CONF | relpath(ROTDIR) }}/{{ head }}bmat_fields_metadata.yaml'
+        - '{{ COMIN_CONF | relpath(ROTDIR) }}/var.yaml'
+        # soca_parameters_diffusion_vt, and soca_parameters_diffusion_hz when present
```
Since it says `when present` should they be in the optional section?

AndrewEichmann-NOAA replied:
@guillaumevernieres Do we want to allow any others to be optional?

DavidHuber-NOAA commented:
@JessicaMeixner-NOAA was this PR wrapped into #4184?

AndrewEichmann-NOAA committed:
Merge branch 'develop' into bugfix/superfluous-yamls

AndrewEichmann-NOAA committed:
Remove unnecessary ensemble background error files

Removed several background error files from the YAML configuration - reverts part of previous merged from develop

AndrewEichmann-NOAA commented:
At this point the only changes are to the `gfs` archiving, which was included for completeness and addresses only a hypothetical issue. Probably a candidate to be closed unmerged.

DavidHuber-NOAA commented:
Closing for now.  This can be revisited if needed at a later time if needed.

JessicaMeixner-NOAA commented:
> @JessicaMeixner-NOAA was this PR wrapped into #4184?

yes.
